### PR TITLE
Etsy scopes

### PR DESCRIPTION
**Pull request by algianotti**
Added a support for Etsy's scopes.

### CODE REVIEW

File: src/OAuth/OAuth1/Service/Etsy.php [hunk 1]
```diff
@@ -27,12 +52,19 @@ public function __construct(
         }
     }
 
+    public function setScopes($requested_scopes = array())
+    {
+        if ($requested_scopes) {
```
Comment (elliotchance) [inline src/OAuth/OAuth1/Service/Etsy.php]: This logic doesn't make sense. The default argument is no scopes, but if there are no scopes nothing will actually change.

Is it valid to have no scopes? If so, how would you set that? It should look something like this:

```php
if ($requested_scopes) {
    $this->scopes = '?scope=' . implode('%20', $requested_scopes);
} else {
    $this->scopes = '';
}
```

File: src/OAuth/OAuth1/Service/Etsy.php [hunk 2]
```diff
@@ -27,12 +52,19 @@ public function __construct(
...
+        }
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getRequestTokenEndpoint()
     {
-        return new Uri($this->baseApiUri . 'oauth/request_token');
+        return new Uri($this->baseApiUri . 'oauth/request_token' . $this->scopes);
```
Comment (elliotchance) [inline src/OAuth/OAuth1/Service/Etsy.php]: There needs to be unit tests for zero, one and multiple scopes set correctly on the Uri.

File: src/OAuth/OAuth1/Service/Etsy.php [hunk 3]
```diff
@@ -27,12 +52,19 @@ public function __construct(
         }
     }
 
+    public function setScopes($requested_scopes = array())
```
Comment (elliotchance) [inline src/OAuth/OAuth1/Service/Etsy.php]: Please use camelCase for all variables and methods - like `$requestedScopes`.

**Comment by elliotchance** [comment]
Please resolve merge conflicts
